react/src/Ipni.js

```diff
@@ -0,0 +1,562 @@
+/* global BigInt */
+import {useQuery} from "@apollo/react-hooks";
+import {
+    IpniAdEntriesCountQuery,
+    IpniAdEntriesQuery,
+    IpniAdQuery,
+    IpniProviderInfoQuery,
+    RetrievalLogsCountQuery, RetrievalLogsListQuery,
+} from "./gql";
+import moment from "moment";
+import React, {useEffect, useState} from "react";
+import {ExpandableJSObject, PageContainer, ShortCIDTail, ShortPeerID} from "./Components";
+import {Link, useNavigate, useParams} from "react-router-dom";
+import {dateFormat, durationNanoToString} from "./util-date";
+import {TimestampFormat} from "./timestamp";
+import './Ipni.css'
+import {Pagination} from "./Pagination";
+import {humanFileSize} from "./util";
+import {addClassFor} from "./util-ui";
+import bezier2Img from "./bootstrap-icons/icons/bezier2.svg";
+import {getDealStatus} from "./RetrievalLogDetail";
+import closeImg from "./bootstrap-icons/icons/x-circle.svg";
+
+const basePath = '/ipni'
+const indexerHost = 'cid.contact'
```

Comment:
It would be better we don't hardcode this value. A config variable would be best. For I think we can start with just one indexer with defaulting to cid.contact.